.build/build.ps1

```diff
@@ -0,0 +1,34 @@
+<#
+.DESCRIPTION
+Builds Kaitai Struct C++ runtime library and unit tests
+
+Requires:
+- MSVC native tools installed and available in the command prompt
+- cmake/ctest available (normally installed with MSVC native tools)
+- GTest installed, path passed in `-GTestPath`
+#>
+
+[CmdletBinding()]
+param (
+    [Parameter(Mandatory=$true)]
+    [string] $GTestPath
+)
+
+# Standard boilerplate
+Set-StrictMode -Version Latest
+$ErrorActionPreference = "Stop"
+$PSDefaultParameterValues['*:ErrorAction']='Stop'
+
+# Go to repo root
+$repoRoot = (Resolve-Path "$PSScriptRoot\..").Path
```

Comment:
If I see correctly, this doesn't go to the repo root. In GitHub Actions workflows, the default working directory is already the repo root (and the `.github/workflows/build.yml` workflow doesn't say otherwise), and mere invocation of a script as `.build\build.ps1` doesn't change the working directory (and the script is also evaluated in the context of the repo root). So if you go to the parent directory anyway, you end up in a parent directory of the repo root, whatever that is.

Reply:
Sorry, I got it wrong here. Still, I think specifying paths relative to the script location like this is confusing and adds unnecessary complexity to all scripts. In general, programs usually do not have this behavior of hardcoding paths of stuff they're working with relative to where they're located. Much more common (and also predictable and flexible) is to use the current working directory to expect certain files/paths to exist (for example `Makefile`). Or, even more flexible and explicit is to simply require the paths to work with to be passed as command-line arguments (I don't personally think we need it here, but it's arguably the clearest option).

I see that your motive probably was to make these two usages equivalent:

```bash
PS C:\temp\kaitai_struct\runtime\cpp_stl> .\.build\build.ps1
```

```bash
PS C:\temp\kaitai_struct\runtime\cpp_stl> cd .build
PS C:\temp\kaitai_struct\runtime\cpp_stl\.build> .\build.ps1
```

but I don't think it's needed. As far as I'm concerned, just expecting that the script will be run from the repo root directory is OK and also easy to understand.

Reply:
You're correct about the purpose, and, ultimately, it's absolutely a matter of personal preference.

I'm obviously biased, but from my experience I've found that having in-repo scripts which work without relying on current working directory being set to anything specific is less error-prone and raises fewer questions — both from humans and from various CI implementations which might want to start at some specific location or limit your ability to just launch one script rather than do `cd somewhere && path/to/script`.

To certain extent, you probably see that 75% of the point of existence of this script in the first place is cmake CLI interface that is indeed highly dependent on current working dir, forcing you to do these weird invocations like `cd build; cmake ..`

If you strongly oppose that, sure, I'll remove it.

Reply:
> If you strongly oppose that, sure, I'll remove it.

My stance on this is actually quite neutral; you're right that it comes down to personal preference. From my standpoint, it's just an unnecessary operation in the code that may potentially contain a bug - which in the case of the shell script it actually does, so let me point it out...